fms2_io/netcdf_io.F90

```diff
@@ -149,6 +150,11 @@ module netcdf_io_mod
   character (len=20) :: time_name
   type(dimension_information) :: bc_dimensions !<information about the current dimensions for regional
                                                !! restart variables
+  logical :: use_collective = .false. !< Flag indicating if we should open the file for collective input
+                                      !! this should be set to .true. in the user application if they want
+                                      !! collective reads (put before open_file())
+  integer :: TileComm=MPI_COMM_NULL   !< MPI communicator used for collective reads.
+                                      !! To be replaced with a real communicator at user request
```

Comment:
You define a variable MPP_COMM_NULL in mpp.F90 as was done for [MPP_INFO_NULL](https://github.com/NOAA-GFDL/FMS/blob/main/mpp/mpp.F90#L1328-L1335) and make it [public](https://github.com/NOAA-GFDL/FMS/blob/main/mpp/mpp.F90#L199).  This way we can keep the MPI layer confined to mpp.